crates/btc-relay/src/lib.rs

```diff
@@ -483,6 +481,53 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Check called by issue/redeem/replace/refund to check the payment
+    pub fn _verify_and_validate_transaction(
```

Comment:
I think it's enough to have one of those verify_and_validate functions since you are submitting two `None` arguments on issue anyway. Might as well submit more `None` arguments and have only a single function.

Reply:
It's 2 additional arguments, and another type for the op_return.

Is it an option to change the `verify_and_validate_transaction` and `validate_transaction` dispatchable functions? I would remove the txid argument from the former (it's redundant due to the raw_tx argument), and change the op_return type to Option<H256>. Then we would only need 1 additional argument here instead of 2, and we wouldn't need to do a H256-to-byte-vec conversion in the individual crates

Reply:
Yes, that sounds very good!